### PR TITLE
Jakarta Support

### PR DESCRIPTION
**Pull request by nebhale**
This change updates the project to support both the javax and jakarta namespaces for JEE projects.  It does this by creating two separate Maven projects, one for each, which are then shaded into a single JAR used by everyone.  The contents of `META-INF/services` and `META-INF/spring.factories` route to the correct implementation preventing class loading problems.

### CODE REVIEW

File: java-buildpack-client-certificate-mapper-jakarta/pom.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="
+                http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.cloudfoundry</groupId>
+        <artifactId>java-buildpack-client-certificate-mapper-parent</artifactId>
+        <version>2.0.0.BUILD-SNAPSHOT</version>
```
Comment (anthonydahanne) [inline java-buildpack-client-certificate-mapper-jakarta/pom.xml]: why keeping this naming `x.y.z.BUILD-SNAPSHOT` ?
Aren't most up to date spring projects using `x.y.z-SNAPSHOT` ?

File: java-buildpack-client-certificate-mapper-jakarta/src/main/java/org/cloudfoundry/router/jakarta/ClientCertificateMapper.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * A Servlet {@link Filter} that translates the {@code X-Forwarded-Client} HTTP header to the {@code javax.servlet.request.X509Certificate} Servlet attribute.  This implementation handles both
+ * multiple headers as well as the <a href=https://tools.ietf.org/html/rfc7230#section-3.2.2>RFC 7230</a> comma delimited equivalent.
+ */
+final class ClientCertificateMapper implements Filter {
+
+    static final String ATTRIBUTE = "javax.servlet.request.X509Certificate";
```
Comment (anthonydahanne) [inline java-buildpack-client-certificate-mapper-jakarta/src/main/java/org/cloudfoundry/router/jakarta/ClientCertificateMapper.java]:
```suggestion
    static final String ATTRIBUTE = "jakarta.servlet.request.X509Certificate";
```

File: java-buildpack-client-certificate-mapper-jakarta/src/main/java/org/cloudfoundry/router/jakarta/ClientCertificateMapper.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * A Servlet {@link Filter} that translates the {@code X-Forwarded-Client} HTTP header to the {@code javax.servlet.request.X509Certificate} Servlet attribute.  This implementation handles both
```
Comment (anthonydahanne) [inline java-buildpack-client-certificate-mapper-jakarta/src/main/java/org/cloudfoundry/router/jakarta/ClientCertificateMapper.java]:
```suggestion
 * A Servlet {@link Filter} that translates the {@code X-Forwarded-Client} HTTP header to the {@code jakarta.servlet.request.X509Certificate} Servlet attribute.  This implementation handles both
```

File: java-buildpack-client-certificate-mapper-jakarta/pom.xml (hunk 1 above)
Comment (anthonydahanne) [inline java-buildpack-client-certificate-mapper-jakarta/pom.xml]:
```suggestion
        <version>2.0.0-SNAPSHOT</version>
```

File: java-buildpack-client-certificate-mapper-javax/pom.xml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="
+                http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.cloudfoundry</groupId>
+        <artifactId>java-buildpack-client-certificate-mapper-parent</artifactId>
+        <version>2.0.0.BUILD-SNAPSHOT</version>
```
Comment (anthonydahanne) [inline java-buildpack-client-certificate-mapper-javax/pom.xml]:
```suggestion
        <version>2.0.0-SNAPSHOT</version>
```

File: java-buildpack-client-certificate-mapper/pom.xml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="
+                http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.cloudfoundry</groupId>
+        <artifactId>java-buildpack-client-certificate-mapper-parent</artifactId>
+        <version>2.0.0.BUILD-SNAPSHOT</version>
```
Comment (anthonydahanne) [inline java-buildpack-client-certificate-mapper/pom.xml]:
```suggestion
        <version>2.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 6]
```diff
@@ -23,119 +23,94 @@
...
-    </parent>
-
     <groupId>org.cloudfoundry</groupId>
-    <artifactId>java-buildpack-client-certificate-mapper</artifactId>
-    <name>Cloud Foundry Client Certificate Mapper</name>
-    <version>1.12.0.BUILD-SNAPSHOT</version>
-    <packaging>jar</packaging>
-    <url>https://github.com/cloudfoundry/java-buildpack-client-certificate-mapper</url>
+    <artifactId>java-buildpack-client-certificate-mapper-parent</artifactId>
+    <version>2.0.0.BUILD-SNAPSHOT</version>
```
Comment (anthonydahanne) [inline pom.xml]:
```suggestion
    <version>2.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 7]
```diff
@@ -23,119 +23,94 @@
...
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>3.2.2</version>
+                    <configuration>
+                        <runOrder>random</runOrder>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.jfrog.buildinfo</groupId>
+                    <artifactId>artifactory-maven-plugin</artifactId>
```
Comment (anthonydahanne) [inline pom.xml:91]: I don't think this plugin utilization is relevant / needed anymore.
With a slight change to [deploy.sh](https://github.com/cloudfoundry/java-buildpack-client-certificate-mapper/pull/8/files#diff-d498a12aa774941c9553350f6bc684a7bb2a70d6b6a2c1fb0619dd90907e8d4dR10) and the [concourse pipeline](https://github.com/pivotal-cf/java-experience-concourse/pull/23/files#diff-d7c0f3daa68586740aac7dfbb56cb7d1fb0979b644700ea0aeaf260e84dd4ea1R31) , we conform to other Java experience projects and don't rely on plugins not coming from central.

**Review by anthonydahanne — CHANGES_REQUESTED** [review]
Thanks so much @nebhale for simplifying everything about Jakarta support!

Since we now deploy to Maven Central, we'll need a bit more XML to add developers, javadoc, licence, url and scm (like I did in the other [PR](https://github.com/cloudfoundry/java-buildpack-client-certificate-mapper/pull/8/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R38))
But we'll add that with David when we test.

Please let me know about the PR comments I left! I'm curious to know your take on those.
Thanks again!

File: java-buildpack-client-certificate-mapper-jakarta/pom.xml (hunk 1 above)
Comment (nebhale) [reply java-buildpack-client-certificate-mapper-jakarta/pom.xml]: Feel free to change this as you choose.  I did the minimum change necessary to get this PR into place.

File: pom.xml (hunk 7 above)
Comment (nebhale) [reply pom.xml:91]: Feel free to remove as you desire.

**Comment by nebhale** [comment]
Nudge @anthonydahanne.

**Review by pivotal-david-osullivan — APPROVED** [review]

**Pull request merged**